<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<!DOCTYPE bugzilla SYSTEM "https://bugs.eclipse.org/bugs/page.cgi?id=bugzilla.dtd">

<bugzilla version="4.4.7"
          urlbase="https://bugs.eclipse.org/bugs/"
          
          maintainer="webmaster@eclipse.org"
>

    <bug>
          <bug_id>424628</bug_id>
          
          <creation_ts>2013-12-24 00:30:00 -0500</creation_ts>
          <short_desc>[1.8][compiler] Multiple method references to inherited method throws LambdaConversionException</short_desc>
          <delta_ts>2014-01-14 09:42:55 -0500</delta_ts>
          <reporter_accessible>1</reporter_accessible>
          <cclist_accessible>1</cclist_accessible>
          <classification_id>2</classification_id>
          <classification>Eclipse</classification>
          <product>JDT</product>
          <component>Core</component>
          <version>4.3.1</version>
          <rep_platform>PC</rep_platform>
          <op_sys>Windows 7</op_sys>
          <bug_status>RESOLVED</bug_status>
          <resolution>FIXED</resolution>
          
          
          <bug_file_loc></bug_file_loc>
          <status_whiteboard></status_whiteboard>
          <keywords></keywords>
          <priority>P3</priority>
          <bug_severity>normal</bug_severity>
          <target_milestone>BETA J8</target_milestone>
          
          
          <everconfirmed>1</everconfirmed>
          <reporter name="Kirk Loretz">kirk_loretz</reporter>
          <assigned_to name="Srikanth Sankaran">srikanth_sankaran</assigned_to>
          <cc>jarthana</cc>
    
    <cc>shankhba</cc>
    
    <cc>srikanth_sankaran</cc>
    
    <cc>stephan.herrmann</cc>
          
          <votes>0</votes>

      

      

      

          <comment_sort_order>oldest_to_newest</comment_sort_order>  
          <long_desc isprivate="0" >
    <commentid>2344771</commentid>
    <comment_count>0</comment_count>
    <who name="Kirk Loretz">kirk_loretz</who>
    <bug_when>2013-12-24 00:30:56 -0500</bug_when>
    <thetext>I&apos;m running:
JDK 1.8.0 b120 64-bit
Eclipse 4.3.1
Eclipse Java Development Tools Patch for Java 8 Support (BETA) 1.0.0.v20131223-0103_BETA_JAVA8

public class MainTest {
    public static interface Consumer&lt;T&gt; {
        void accept(T t);
    }
    
    public static class Base {
        public void method () {}
    }
    public static class Foo extends Base {}
    public static class Bar extends Base {}

    public static void main (String[] args) {
        Consumer&lt;Foo&gt; foo = Foo::method;
        Consumer&lt;Bar&gt; bar = Bar::method;
    }
}

The above class compiles fine with Eclipse, but when I go to run it I get the following error:

Exception in thread &quot;main&quot; java.lang.BootstrapMethodError: call site initialization exception
    at java.lang.invoke.CallSite.makeSite(CallSite.java:328)
    at java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:296)
    at MainTest.main(MainTest.java:19)
Caused by: java.lang.invoke.LambdaConversionException: Invalid receiver type class MainTest$Foo; not a subtype of implementation type class MainTest$Bar
    at java.lang.invoke.AbstractValidatingLambdaMetafactory.validateMetafactoryArgs(AbstractValidatingLambdaMetafactory.java:221)
    at java.lang.invoke.LambdaMetafactory.metafactory(LambdaMetafactory.java:303)
    at java.lang.invoke.CallSite.makeSite(CallSite.java:289)
    ... 2 more

To get the class to run without an error via Eclipse, I can either implement method() in one of the subclasses or change both method references in main to Base::method. However, neither of these changes is needed to compile and run using JDK8.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>2344777</commentid>
    <comment_count>1</comment_count>
    <who name="shankha banerjee">shankhba</who>
    <bug_when>2013-12-24 02:32:02 -0500</bug_when>
    <thetext>Working on it.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>2346290</commentid>
    <comment_count>2</comment_count>
    <who name="Stephan Herrmann">stephan.herrmann</who>
    <bug_when>2014-01-06 20:10:02 -0500</bug_when>
    <thetext>Pleases see the disabled test NullTypeAnnotationTest._testBug424637_comment3() for another situation throwing the same exception.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>2349414</commentid>
    <comment_count>3</comment_count>
    <who name="Srikanth Sankaran">srikanth_sankaran</who>
    <bug_when>2014-01-14 09:02:50 -0500</bug_when>
    <thetext>(In reply to Stephan Herrmann from comment #2)
&gt; Pleases see the disabled test
&gt; NullTypeAnnotationTest._testBug424637_comment3() for another situation
&gt; throwing the same exception.

This is not related to this bug, but to https://bugs.eclipse.org/bugs/show_bug.cgi?id=406744. I have added a note there.</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>2349434</commentid>
    <comment_count>4</comment_count>
    <who name="Srikanth Sankaran">srikanth_sankaran</who>
    <bug_when>2014-01-14 09:42:35 -0500</bug_when>
    <thetext>Thanks for the test Kirk.

Problem caused by bad bug fix to https://bugs.eclipse.org/bugs/show_bug.cgi?id=423684.

Fixed via: http://git.eclipse.org/c/jdt/eclipse.jdt.core.git/commit/?h=BETA_JAVA8&amp;id=7186c730d487753734b0547672e890b50fededb4</thetext>
  </long_desc>
      
      

    </bug>

</bugzilla>